### ISSUE TITLE
server ignores request to gzip data

### ISSUE THREAD

**Issue by tmeinlschmidt** [prysmaticlabs/prysm#14593]
### Describe the bug

version: beacon-chain:v5.0.4

was checking tcpdump as we had some issues and found following

````
(GET /eth/v1/beacon/blob_sidecars/10252744?indices=0&indices=1&indices=2&indices=3&indices=4 HTTP/1.1
Host: beacon:7028
User-Agent: Go-http-client/1.1
Accept: application/json
Accept-Encoding: gzip
````
response
````
Content-Type: application/json
Vary: Origin
Vary: Origin
Date: Tue, 29 Oct 2024 17:06:42 GMT
Transfer-Encoding: chunked

142720
{"data":[{"index":"0","blob":"0x3e0001fbfc...
````

so it ignored Accept-Encoding header and sending only plain json. Would be great if beacon can handle this correctly.

### Platform(s)

Linux (x86)

### What version of Prysm are you running? (Which release)

5.0.4

**Comment by james-prysm** [comment]
I don't believe this is a required feature of https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlobSidecars. Do other clients support this encoding?

**Comment by tmeinlschmidt** [comment]
rpcdaemon (as server) and erigon (as client) works just fine and compress data. 

request to beacon came from base/ptimism chain

**Comment by prestonvanloon** [comment]
Seems like a reasonable request to have the beacon chain return gzip encoded payloads when requested for the beacon API.

**Comment by rose2221** [comment]
i'd like to work on this issue

**Comment by rkapka** [comment]
Thanks @rose2221 , I assigned it to you

**Comment by DarkLord017** [comment]
Is the issue still open , I would like to work on it @rkapka

**Comment by abuzaforfagun** [comment]
Is this still open? would like to contribute.
